src/request_queue.js

```diff
@@ -197,26 +206,45 @@ export class RequestQueue {
         checkParamOrThrow(queueName, 'queueName', 'Maybe String');
         checkParamOrThrow(clientKey, 'clientKey', 'String');
 
-        if (!clientKey) throw new Error('Parameter "clientKey" must be nonempty string!');
+        if (!clientKey) throw new Error('Parameter "clientKey" must be a non-empty string!');
 
         this.clientKey = clientKey;
         this.queueId = queueId;
         this.queueName = queueName;
+
+        // Contains a cached list of request IDs from the head of the queue,
+        // as obtained in the last query. Both key and value is the request ID.
         this.queueHeadDict = new ListDictionary();
-        this.requestIdsInProgress = {};
-        this.inProgressCount = 0;
         this.queryQueueHeadPromise = null;
 
+        // Contains a list of all requests that are currently being handled.
+        // Keys are request IDs, values are true.
+        this.inProgress = {};
+
+        // Contains a list of recently handled requests. It is used to avoid inconsistencies
+        // caused by delays in the underlying DynamoDB storage.
+        // Keys are request IDs, values are true.
+        this.recentlyHandled = new LruCache({ maxLength: RECENTLY_HANDLED_CACHE_SIZE });
+
         // We can trust these numbers only in a case that queue is used by a single client.
         // This information is returned by getHead() under the hadMultipleClients property.
         this.assumedTotalCount = 0;
         this.assumedHandledCount = 0;
 
-        // Caching requests to avoid duplicite addRequest() calls.
+        // Caching requests to avoid redundant addRequest() calls.
         // Key is computed using getRequestId() and value is { id, isHandled }.
+        // TODO: We could extend the functionality of caching to improve performance
+        //       of other operations too, such as fetchNextRequest().
         this.requestsCache = new LruCache({ maxLength: MAX_CACHED_REQUESTS });
     }
 
+    /**
+     * @ignore
+     */
+    inProgressCount() {
+        return Object.keys(this.inProgress).length;
```

Comment:
With a `Set`, this becomes just `this.inProgress.size`

Reply:
done